public/static/js/distros.js

```diff
@@ -195,24 +158,73 @@ mciModule.controller('DistrosCtrl', function($scope, $window, $location, $anchor
     var distroHash = $location.hash();
     if (distroHash) {
       // If the distro exists, load it.
-      var distro = $scope.getDistroById(distroHash);
-      if (distro) {
-    $scope.activeDistro = distro;
-    return;
+      if (distroHash == newId && $scope.tempDistro) {
```

Comment:
nit: use === and !== so that javascript doesn't do type conversions before comparing the values